### PR TITLE
Ensure issue.Poster is loaded in mailIssueCommentToParticipants

### PR DESCRIPTION
**Pull request by zeripath**
Previous code could potentially dereference nil - this PR ensures
that the poster is loaded before dereferencing it.

Fixes issue reported on Discord by drewG 

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

File: models/issue_mail.go [hunk 1]
```diff
@@ -39,11 +39,11 @@ func mailIssueCommentToParticipants(e Engine, issue *Issue, doer *User, content
 
 	// In case the issue poster is not watching the repository and is active,
 	// even if we have duplicated in watchers, can be safely filtered out.
-	poster, err := getUserByID(e, issue.PosterID)
+	issue.loadPoster(e)
```
Comment (adelowo) [inline models/issue_mail.go]: It doesn't seem like you are checking the returned error

Comment (zeripath) [reply models/issue_mail.go]: Oops. Sorry.

**Comment by zeripath** [comment]
@adelowo is my change ok? (I really think we need to get this approved quickly and before we release v1.7.1 as every person who is on 1.7 and has ENABLE_NOTIFY_MAIL set to `true` will be experiencing server 500s.)

**Review by techknowlogick — APPROVED** [review]

**Review by adelowo — APPROVED** [review]

**Comment by zeripath** [comment]
Of course I then rebase and run my magical #5893 on it and notice another dereference bug... 

The repo needs to be loaded too in v1.7 that is. I'll put the fix in to the backport and open another one on master - which is I think protected by these things being loaded higher up but we should ensure that they're there at this level I think.

**Pull request merged**